frozendict/tests.py

```diff
@@ -0,0 +1,131 @@
+            *args, **kwargs
+        )
+
+
+def setitem(dct, key, value):
+    dct[key] = value
+
+
+class TestFrozenDict(BaseTestCase):
+    def test_assignment_fails(self):
+        self.assertRaisesAssignmentError(setitem, frozendict(), 'key', 'value')
+
+    def test_init(self):
+        frozendict()
+
+    def test_init_with_keys(self):
+        fd = frozendict(a=0, b=1)
+        self.assertMappingEqual(fd, dict(a=0, b=1))
+
+    def test_init_with_dict(self):
+        fd = frozendict(dict(a=0, b=1))
+        self.assertMappingEqual(fd, dict(a=0, b=1))
+
+    def test_init_with_dict_and_keys(self):
+        fd = frozendict(dict(a=0, b=1), b=2, c=3)
+        self.assertMappingEqual(fd, dict(a=0, b=2, c=3))
+
+    def test_getitem(self):
+        fd = frozendict(key='value')
+        self.assertEqual(fd['key'], 'value')
+
+    def test_copy(self):
+        fd = frozendict(a=0, b=1)
+        copied = fd.copy()
+        self.assertIs(type(copied), type(fd))
+        self.assertMappingEqual(copied, fd)
+
+    def test_copy_with_keys(self):
+        fd = frozendict(a=0, b=1)
+        copied = fd.copy(b=2, c=3)
+        self.assertIs(type(copied), type(fd))
+        self.assertMappingEqual(copied, dict(a=0, b=2, c=3))
+
+    def test_iter(self):
+        fd = frozendict(a=0, b=1)
+        keys = {key for key in fd}
+        self.assertSetEqual(keys, {'a', 'b'})
+
+    def test_len(self):
+        fd = frozendict(a=0, b=1)
+        self.assertEqual(len(fd), 2)
+
+    def test_repr(self):
+        fd = frozendict(a=0, b=1)
+        r = repr(fd)
+        self.assertTrue(r.startswith('<frozendict '))
+        self.assertTrue(r.endswith('>'))
+
+    def test_hash(self):
+        fd = frozendict(a=0, b=1)
+        hash(fd)
+
+
+class TestFrozenOrderedDict(BaseTestCase):
+    def test_assignment_fails(self):
+        self.assertRaisesAssignmentError(setitem, FrozenOrderedDict(), 'key', 'value')
+
+    def test_init(self):
+        FrozenOrderedDict()
+
+    def test_init_with_keys(self):
+        fd = FrozenOrderedDict(a=0, b=1)
+        self.assertMappingEqual(fd, dict(a=0, b=1))
+
+    def test_init_with_dict(self):
+        fd = FrozenOrderedDict(dict(a=0, b=1))
+        self.assertMappingEqual(fd, dict(a=0, b=1))
+
+    def test_init_with_dict_and_keys(self):
+        fd = FrozenOrderedDict(dict(a=0, b=1), b=2, c=3)
+        self.assertMappingEqual(fd, dict(a=0, b=2, c=3))
+
+    def test_getitem(self):
+        fd = FrozenOrderedDict(key='value')
+        self.assertEqual(fd['key'], 'value')
+
+    def test_copy(self):
+        fd = FrozenOrderedDict(a=0, b=1)
+        copied = fd.copy()
+        self.assertIs(type(copied), type(fd))
+        self.assertMappingEqual(copied, fd)
+
+    def test_copy_with_keys(self):
+        fd = FrozenOrderedDict(a=0, b=1)
+        copied = fd.copy(b=2, c=3)
+        self.assertIs(type(copied), type(fd))
+        self.assertMappingEqual(copied, dict(a=0, b=2, c=3))
+
+    def test_iter(self):
```

Comment:
there is a 1/16th chance that this test will pass with frozendict. Please increase the test data size to reduce the chance to a more reasonable value